2.0.md

```diff
@@ -0,0 +1,895 @@
+
+For example, for our bike sharing app some activities for the bike stream could be
+
+1. ride started
+2. ride ended
+3. maintenance requested
+4. moved to new location
+
+Some activities for the customer could be
+
+1. purchased daily pass
+2. started ride
+3. purchased yearly subscription
+4. submitted maintenance request
+5. viewed web page
+6. opened ride app
+7. viewed bike availability
+
+Once the activities have been identified then it's usually fairly straightforward to find which source table(s) will be needed. The only requirement is that we can identify an entity, a relevant activity, and a timestamp. 
+
+<br>
+<br>
+
+
+### SQL Transformations
+
+SQL Transformations are short SQL queries that map from source data to the activity stream format. They are typically fairly small and easy to write. 
+
+For example, this is how **completed_order** is built from Shopify data in Redshift
+
+```sql
+SELECT
+  o.id AS activity_id
+  , o.processed_at AS ts
+  , NULL     AS anonymous_customer_id
+  , o.email  AS customer
+
+  , 'completed_order' AS activity
+  , object (
+    'discount_code', d.code,
+    'order_name', o.name
+  ) AS feature_json
+
+  , (o.total_price - o.total_discounts) AS revenue_impact
+
+FROM shopify.order AS o
+LEFT JOIN shopify.order_discount_code d
+  ON (d.order_id = o.id)
+
+WHERE
+  o.cancelled_at is NULL
+  and o.email is not NULL
+  and o.email <> ''
+```
+
+This SQL is pretty straightforward
+
+1. the **completed_order** activity is defined independently — it doesn't have to join with any other tables. We just need to find the customer's unique identifier (in this case email)
+2. activities generally conform to well-known business processes, so the data is frequently close to the desired format already
+
+In practice nearly all transformation scripts are less than 30 lines and one or two joins. 
+
+In addition, there's no reason that transformation scripts and activities have to be 1:1. For example, a 'received_email' activity could be defined from source tables from multiple systems. Or a single transformation can create multiple activities.
+
+
+<br>
+<br>
+
+
+### Building an activity stream
+
+Companies maintaining data models in their warehouse run scheduled tasks to keep all tables up to date. In practice, this is done with a periodically running scheduled task (using a tool like dbt) that carefully manages a graph of table dependencies.
+
+The activity schema approach is effectively the same, only without the layers of dependencies. A common approach is to create a single SQL query per activity desired, and append the results of each query to the activity stream table. 
+
+This approach also means that incremental updates to the activity stream are straightforward (identify all activity rows created by the given transformation, find the maximum timestamp, and insert rows with a newer timestamp).
+
+Building an activity schema implementation requires these steps
+
```

Comment:
I would recommend linking to the Implementation.md here and removing the bullets below.